File: 백준/Silver/5567. 결혼식/결혼식.py
import sys
input = sys.stdin.readline
from collections import deque

n = int(input())
m = int(input())
graph = [ [0] * (n+1) for _ in range(n+1)]
visited = [0 for _ in range(n+1)]

for i in range(m):
    a,b = map(int,input().split())
    graph[a].append(b)
    graph[b].append(a)

def bfs(x):
    q = deque()
    visited[x] = 1
    q.append(x)
    while q:
        a = q.popleft()
        for i in graph[a]:
            if visited[i] == 0:
                q.append(i)
                visited[i] = visited[a] + 1

bfs(1)
result = 0
for i in range(2,n+1):
    # 본인이거나 친구거나, 친구의 친구거나 경우의 수가 최대 3개
    if visited[i] < 4 and visited[i] != 0:
        result += 1
print(result)

File: 백준/Silver/5567. 결혼식/결혼식.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const n = Number(input[0])
const m = Number(input[1])
const graph = Array.from({ length: n + 1 }, () => [])
const visited = Array.from({ length: n + 1 }, () => false)
const target = input.slice(2, 2 + m).map((row) => row.split(" ").map(Number))
target.forEach(([a, b]) => {
  graph[a].push(b)
  graph[b].push(a)
})
let cnt = 0

function bfs(start){
  const queue = [[start,0]]
  visited[start] = true
  let index = 0
  
  while(queue.length>index){
    const [cur,depth] = queue[index++]

    for(let next of graph[cur]){
      if(!visited[next]){
        if(depth+1 <= 2 ){
          cnt +=1
        }
        queue.push([next,depth+1])
        visited[next] = true

      }
    }
  }
}

bfs(1)

console.log(cnt)